be/src/storage/delta_writer.h

```diff
@@ -123,6 +127,8 @@ class DeltaWriter {
     const TabletSchema* _tablet_schema;
 
     std::unique_ptr<FlushToken> _flush_token;
+    FlushChunkState _flush_chunk_state = FlushChunkState::unknown;
+    FlushChunkState _curr_flush_chunk_state = FlushChunkState::unknown;
```

Comment:
It seems putting state in rowset writer rather than delta_writer can simply things a lot

Reply:
good idea, changed.